dietpi/dietpi-software

```diff
@@ -4332,6 +4339,223 @@ _EOF_
+    admin_password: "dietpi-lemmy"
+    site_name: "My DietPi Lemmy Instance"
+  }
+  ## Disable this if not a federated instance
+  tls_enabled: true
+}
+_EOF_
+			## Backend Service
+			# Pictrs options: https://git.asonix.dog/asonix/pict-rs/
+			G_EXEC mkdir -p "$lemmy_userdata/pictrs/sled-repo"
+			G_EXEC mkdir -p "$lemmy_userdata/pictrs/store"
+			cat << _EOF_ > /etc/systemd/system/lemmy.service
+[Unit]
+Description=Lemmy - A link aggregator for the fediverse
+After=network.target
+
+[Service]
+User=lemmy
+ExecStart=$lemm_server_path/lemmy_server
+Environment=LEMMY_CONFIG_LOCATION="$lemmy_userdata/lemmy.hjson"
+Environment=PICTRS_PATH=$lemmy_userdata/pictrs
+Environment=PICTRS__REPO__PATH=$lemmy_userdata/pictrs/sled-repo
+Environment=PICTRS__STORE__PATH=$lemmy_userdata/pictrs/store
+Environment=PICTRS_ADDR=$host_bind:$port_pictrs
+Restart=on-failure
+
+# Hardening
+ProtectSystem=yes
+PrivateTmp=true
+MemoryDenyWriteExecute=true
+NoNewPrivileges=true
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+			## UI Service - https://github.com/LemmyNet/lemmy-ui#configuration
+			cat << _EOF_ > /etc/systemd/system/lemmy-ui.service
+[Unit]
+Description=Lemmy UI - Web frontend for Lemmy
+After=lemmy.service
+##Before=nginx.service
+
+[Service]
+User=lemmy
+WorkingDirectory=/var/lib/lemmy-ui
+ExecStart=/usr/bin/node dist/js/server.js
+Environment=LEMMY_UI_HOST=$host_broadcast:$port_frontend
+Environment=LEMMY_UI_LEMMY_INTERNAL_HOST=$host_bind:$port_backend
+Environment=LEMMY_UI_LEMMY_EXTERNAL_HOST="$host_domain"
+Environment=LEMMY_UI_EXTRA_THEMES_FOLDER="$lemmy_userdata/extra-themes"
+Environment=LEMMY_UI_HTTPS=true ## Hey, should this be enforced?
+Restart=on-failure
+
+# Hardening
+ProtectSystem=full
+PrivateTmp=true
+NoNewPrivileges=true
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+			## TODO: Upgrade Scripts. FIX THIS SO THAT IT WORKS WITH THE GLOBAL CARGO ENV
+    		cat << _EOF_ > /etc/bashrc.d/lemmy.sh
+#!/bin/dash
+lemmy-update()
+{
+	case "\$1" in
+		l*) ## Lemmy
+			sudo -u '$lemm_user' dash -c '. "$lemm_home/.cargo/env"; cargo install lemmy_server --locked --features embed-pictrs'
+			sudo systemctl restart lemmy
+			;;
+		ui) ## UI
+			sudo -u '$lemm_user' dash -c '\
+. "$lemm_home/.cargo/env"; \
+cd /var/lib/lemmy-ui; \
+git checkout main && \
+git pull --tags && \
+git checkout main && \
+git submodule update && \
+yarn install --pure-lockfile && \
+yarn build:prod;';
+			sudo systemctl restart lemmy-ui
+			;;
+		*) echo 'lemmy-update helper script
+
+usage: lemmy-update <lemmy|ui>
+
+where:
+  lemmy   update and rebuild the backend server (via cargo)
+  ui      update and rebuild the ui (via yarn)
+
+  branch  defaults to 'main'
+		   return 1;;
+	esac
+}
+_EOF_
+
+			# Postgres
+			## We need to tell postgres that we're doing password auth (md5) and not peer.
+			G_EXEC sed -i -r '/^local\s+all\s+all\s+peer/i local   lemmy		   lemmy								   md5' /etc/postgresql/15/main/pg_hba.conf
```

Comment:
1. The PostgreSQL version depends on the Debian version: https://packages.debian.org/postgresql
2. However, it shouldn't be required to alter any config file for this. Something like this should create a user with password authentication:
```sh
if [[ $(sudo -u lemmy psql -tAc "SELECT 1 FROM pg_roles WHERE rolname='lemmy'") != 1 ]]
then
	G_EXEC sudo -u lemmy psql -c "CREATE ROLE synapse WITH LOGIN PASSWORD '$pass_database';"
else
	G_EXEC sudo -u lemmy psql -c "ALTER ROLE synapse WITH PASSWORD '$pass_database';"
fi
```
I think/hope MD5 is deprecated and not the default how PostgreSQL is hashing passwords. MD5 is okay for integrity checks only, not secure at all since decades 😉.

Reply:
Hmm! I tried this at first, but could not login to the database. I'll try it again just in case I overlooked something